reference/extensions/python_requires.rst

```diff
@@ -1,4 +1,304 @@
+
+.. code-block:: bash
+
+    $ conan create .
+    # It will only export the recipe, but NOT create binaries
+    # python-requires do NOT have binaries
+
+
+We can reuse the above recipe functionality declaring the dependency in the ``python_requires``
+attribute and we can access its members using ``self.python_requires["<name>"].module``:
+
+.. code-block:: python
+    
+    from conan import ConanFile
+
+    class Pkg(ConanFile):
+        name = "pkg"
+        version = "0.1"
+        python_requires = "pyreq/0.1"
+
+        def build(self):  
+            v = self.python_requires["pyreq"].module.myvar  # v will be 123
+            f = self.python_requires["pyreq"].module.myfunct()  # f will be 234
+            self.output.info(f"{v}, {f}")
+
+
+.. code-block:: bash
+
+    $ conan create . 
+    ...
+    pkg/0.1: 123, 234
+
+
+Python requires can also use version ranges, and this can be recommended in many cases if those ``python-requires``
+need to evolve over time:
+
+.. code-block:: python
+    
+    from conan import ConanFile
+
+    class Pkg(ConanFile):
+        python_requires = "pyreq/[>=1.0 <2]"
+
+
+It is also possible to require more than 1 ``python-requires`` with ``python_requires = "pyreq/0.1", "other/1.2"``
+
+Extending base classes
+----------------------
+
+A common use case would be to declare a base class with methods we want to reuse in several
+recipes via inheritance. We'd write this base class in a python-requires package: 
+
+.. code-block:: python
+
+    from conan import ConanFile
+
+    class MyBase:
+        def source(self):
+            self.output.info("My cool source!")
+        def build(self):
+            self.output.info("My cool build!")
+        def package(self):
+            self.output.info("My cool package!")
+        def package_info(self):
+            self.output.info("My cool package_info!")
+
+    class PyReq(ConanFile):
+        name = "pyreq"
+        version = "0.1"
+        package_type = "python-require"
+
+
+And make it available for reuse with:
+
+.. code-block:: bash
+
+    $ conan create .
+
+
+Note that there are two classes in the recipe file:
+
+ * ``MyBase`` is the one intended for inheritance and doesn't extend ``ConanFile``.
+ * ``PyReq`` is the one that defines the current package being exported, it is the recipe
+   for the reference ``pyreq/0.1``.
+
+
+Once the package with the base class we want to reuse is available we can use it in other
+recipes to inherit the functionality from that base class. We'd need to declare the
+``python_requires`` as we did before and we'd need to tell Conan the base classes to use
+in the attribute ``python_requires_extend``. Here our recipe will inherit from the
+class ``MyBase``:
+
+
+.. code-block:: python
+    
+    from conan import ConanFile
+
+    class Pkg(ConanFile):
+        name = "pkg"
+        version = "0.1"
+        python_requires = "pyreq/0.1"
+        python_requires_extend = "pyreq.MyBase"
```

Comment:
can I use `python_requires_extend = "pyreq.MyBase1", "myreq.MyBase2"`?

Reply:
Yes, but not recommended, read below